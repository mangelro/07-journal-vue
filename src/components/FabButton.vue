import { defineAsyncComponent } from '@vue/runtime-core';
<template>
  <button class="fab circle extra" @click="$emit('on:click',$event)">
  <i>{{icon}}</i>
</button>
</template>

<script>
export default {
    props:{
        icon:{
            type:String,
            default:'save'
        }
    },
    emits:['on:click']
}
</script>

<style>

</style>