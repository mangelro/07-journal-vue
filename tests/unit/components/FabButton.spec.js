import {shallowMount} from '@vue/test-utils'
import FabButton from '@/components/FabButton'

describe('Prueba del componente FabButon ',()=>{

    // let wrapper
    // beforeEach(()=>
    // {
    // })

	test('Debe renderizar el componente correctamente contra el snapshot',()=>{
        const wrapper=shallowMount(FabButton)
		expect(wrapper.html()).toMatchSnapshot()

	}),

    test('Debe mostar el icono por defecto',()=>{
        const wrapper=shallowMount(FabButton)

        expect (wrapper.props('icon')).toBe('save')

    })

})