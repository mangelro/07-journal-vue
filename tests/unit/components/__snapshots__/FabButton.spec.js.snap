// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del componente FabButon  Debe renderizar el componente correctamente contra el snapshot 1`] = `<button class="fab circle extra"><i>save</i></button>`;
